contracts/Content/UniqueContent.sol

```diff
@@ -0,0 +1,173 @@
+        uint256 _tokenId = uniqueAssetInfo[_uniqueId].tokenId;
+        return IContent(uniqueAssetInfo[_uniqueId].contentAddress).uri(_tokenId, _version);
+    }
+
+    /**
+    * @dev Returns the unique asset uri of a specific version
+    * @param _uniqueId uint256 ID of token to query
+    * @param _version version number of token to query
+    */
+    function uri(uint256 _uniqueId, uint256 _version) external view override returns (string memory) {
+        require(_exists(_uniqueId), "Unique Id does not exist");
+        if (_version > uniqueAssetInfo[_uniqueId].version) {
+            _version = uniqueAssetInfo[_uniqueId].version;
+        } 
+        return uniqueAssetInfo[_uniqueId].uniqueAssetUri[_version];
+    }
+
+    /**
+    * @dev Adds a new version of the unique asset
+    * @param _uniqueId uint256 ID of the token to set its uri
+    * @param _uri string URI to assign
+    */
+    function setUniqueUri(uint256 _uniqueId, string memory _uri) external override {
+        require(_exists(_uniqueId), "Unique Id does not exist");
+        require(uniqueAssetInfo[_uniqueId].creator == _msgSender(), "You are not the creator of this token");
+        uniqueAssetInfo[_uniqueId].uniqueAssetUri.push(_uri);
+        uint256 _version = ++uniqueAssetInfo[_uniqueId].version;
+
+        emit UniqueUriUpdated(_uniqueId, _version);
+    }
+
+    /**
+    * @dev Returns the original asset's receiver address and royalty amount for a token sold at a certain sales price
+    * @param _uniqueId uint256 ID of token to query
+    * @param _salePrice price the asset is to be purchased for
+    */
+    function royaltyInfo(uint256 _uniqueId, uint256 _salePrice) external view override returns (address receiver, uint256 royaltyAmount){
+        require(_exists(_uniqueId), "Unique Id does not exist");
+        // grabs the original asset's content contract address and tokenId
+        address originalAddress = uniqueAssetInfo[_uniqueId].contentAddress;
+        uint256 tokenId = uniqueAssetInfo[_uniqueId].tokenId;
+        // calls royaltyInfo() in the original asset's contract
+        (receiver, royaltyAmount) = IERC2981Upgradeable(originalAddress).royaltyInfo(tokenId, _salePrice);
+    }
+
+    /**
+    * @dev Returns an array of receiver addresses and royalty amounts for a token sold at a certain sales price
+    * @param _uniqueId uint256 ID of token to query
+    * @param _salePrice price the asset is to be purchased for
+    */
+    function multipleRoyaltyInfo(uint256 _uniqueId, uint256 _salePrice) external view override returns (address[] memory receivers, uint256[] memory royaltyAmounts) {
+        require(_exists(_uniqueId), "Unique Id does not exist");
+        uint24[] memory rates;
+        // grabs the arrays of receivers and rates for the token
+        (receivers, rates) = _getMultipleRoyalties(_uniqueId);
+        // calculates royaltyAmount for each receiver and places them in an array
```

Comment:
You forgot to add the royalty from the original asset. 

We still need to distribute some royalties to the original developer. We can't assume that the unique asset creator will do that for you. We also can't assume that an exchange will call both "royaltyInfo()" and "multipleRoyaltyInfo()" just to get both new and old royalties